hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/XceiverServerRatis.java

```diff
@@ -107,6 +112,8 @@ private static long nextCallId() {
   // TODO: Remove the gids set when Ratis supports an api to query active
   // pipelines
   private final Set<RaftGroupId> raftGids = new HashSet<>();
+  // pipeline leaders
+  private Map<RaftGroupId, RaftPeerId> leaderIdMap = new HashMap<>();
```

Comment:
This should be a concurrentHashMap as pipeline report and changeLeader can happen at anytime.

Reply:
Yes, I thought about this already and left it HashMap, thinking we would only get leader changed notification reporting new leaderId, so the updated value will always be the same from any member. For the sake of caution making it CHM.